libs/blocks/preflight/panels/accessibility.js

```diff
@@ -0,0 +1,59 @@
+import { html, signal, useEffect } from '../../../deps/htm-preact.js';
+
+const DEF_ICON = 'purple';
+const DEF_DESC = 'Checking...';
+const pass = 'green';
+const fail = 'red';
+
+const content = signal({});
+const altResult = signal({ icon: DEF_ICON, title: 'Image alt value', description: DEF_DESC });
+
+async function checkAlt() {
+  const main = document.querySelector('main');
+  const images = main.querySelectorAll('img');
+  const result = { ...altResult.value };
+  const imagesWithoutAlt = [];
+  images.forEach((img) => {
+    const alt = img.getAttribute('alt');
+    if (!alt || alt.trim() === '') {
+      imagesWithoutAlt.push(img.getAttribute('src'));
+    }
+  });
+  if (!imagesWithoutAlt.length) {
+    result.icon = pass;
+    result.description = 'Reason: All Image are valid';
+  } else {
+    result.icon = fail;
+    result.description = 'Reason: Alt attribute or values are missing for below images on the page';
+  }
+  content.value = imagesWithoutAlt;
+  altResult.value = result;
+  return result.icon;
```

Comment:
How is this used?

Reply:
Thanks for the review @narcis-radu It was a mistake at my side we don't need to return anything here. I have updated the code.
For **content.value** it contains all the images that are missing alt attribute or value to create the grid on modal.
**altResult.value** contains the details of message and icon to show based on checks have passed or failed